gcip/addons/container/config.py

```diff
@@ -0,0 +1,135 @@
+cfg.add_cred_helper("1234567890.dkr.ecr.us-east-1.amazonaws.com", "ecr-login")
+cfg.render()
+```
+
+This will render a Client configuration and dumps it as a json string.
+"""
+import json
+from typing import Any, Dict
+
+DockerConfig = Dict[str, Any]
+
+
+class DockerClientConfig():
+    def __init__(self, config_file_path: str = "$HOME/.docker/", config_file_name: str = "config.json") -> None:
+        """
+        Class which represents a docker client configuration.
+
+        After creating an instance of this class you can add new credential helper or
+        basic authentication settings. Or set a default credential store.
+        The constructor has a limited feature set. If you change either `config_file_path` or `config_file_name`,
+        you should consider using `docker --config` to specify the path to the configuration file.
+
+        Args:
+            config_file_path (str): Filesystem path where to create the docker client directory. Defaults to $HOME/.docker/.
+            config_file_name (str): Docker client configuration filename. Defaults to config.json.
+        """
+        self._config_file_path = config_file_path
+        self._config_file_name = config_file_name
+        self.config: DockerConfig = {}
+
+    def set_creds_store(self, creds_store: str) -> None:
+        """
+        Sets the `credsStore` setting for clients.
+        See [docker login#credentials-store](https://docs.docker.com/engine/reference/commandline/login/#credentials-store)
+
+        Be aware, that if you set the `credsStore` and add creds_helper or
+        username and password authentication, those authentication methods
+        are not used.
+
+        Clients which can authenticate against a registry can handle the credential
+        store itself, mostly you do not want to set the `credsStore`.
+        Use `credsHelpers` instead.
+
+        Args:
+            creds_store (str): Should be the suffix of the program to use
+            (i.e. everything after docker-credential-).
+            For example `osxkeychain`, to use docker-credential-osxkeychain or
+            `ecr-login`, to use docker-crendential-ecr-login
+        """
+        self.config["credsStore"] = creds_store
+
+    def add_cred_helper(self, registry: str, cred_helper: str) -> None:
+        """
+        Adds a Credentials helper `credHelpers` for a registry.
+        See [docker login#credential-helpers](https://docs.docker.com/engine/reference/commandline/login/#credential-helpers)
+
+        Args:
+            registry (str): Name of the container registry to set `creds_helper` for.
+            cred_helper (str): Name of the credential helper to use together with the `registry`.
+        """
+        compose = {
+            registry: cred_helper
+        }
+        if "credsHelpers" not in self.config.keys():
+            self.config["credHelpers"] = compose
+        else:
+            self.config["credHelpers"].update(compose)
+
+    def add_auths(
```

Comment:
You add a single 'auth' to plural 'auths' here:
```suggestion
    def add_auth(
```

Reply:
Oops, that should have been changed.